contributing-tips.rst

```diff
@@ -0,0 +1,142 @@
+Tips for Making Good Contributions
+====================================
+
+.. _small-pr:
+
+Smallest meaningful PR
+------------------------
+
+A PR should normally address one issue. This makes it easier to review, easier to deploy, and easier to roll back in case of a problem. Additionally, the smaller the PR, the less likely it is to create a merge issue.
+
+The exception is when several issues are closely related or can reasonably be worked on together.  In this case, it should be clear by looking at the conversation on the Github issues that the items are related and will be worked on together.
+
+
+.. _descriptive-pr-names:
+
+Descriptive PR names
+----------------------
+
+A PR title should answer the question, "What does this Pull Request do?"
+
+Good PR titles:
+
+- adds a video directive
+- makes navigation buttons responsive
+- swaps placement of nav buttons and file an issue note
+
+Bad PR titles:
+
+- fix issue
+- fix #123
+- collect
+
+.. _small-commits:
+
+Small Atomic Commits
+-----------------------
+
+When working locally, commit often. Don't wait until you have 100 lines of changes across multiple files.
+
+- If you need to copy or move a large section or file, commit that change before also editing it. 
+- If you have to create a new template file based on an existing template file, copy the file in one commit and then work on the changes. This makes it easier to know what you actually did.
```

Comment:
If you have to create a new template file based on an existing one ( just trying to minimize words here )

Reply:
i'll think about it